src/install/dracut-install.c

```diff
@@ -567,7 +567,7 @@ static int resolve_deps(const char *src)
 
                 /* glibc */
                 if (strstr(buf, "cannot execute binary file"))
```

Comment:
> This changes the original intent of [b127294](https://github.com/dracutdevs/dracut/commit/b127294def5efecc27fac730f784f8bf03a5e52d) . I am not able to speculate what problem the original commit was solving.

I can only speculate on a 32-bit system calling `ldd` against a 64-bit library, where this error would be displayed at the beginning and nothing else.

> 
> It would help to understand better what are the conditions for ldd to output "cannot execute binary file". Are they any conditions where the output matches "cannot execute binary file" but does NOT match "cannot execute binary file: Exec format error" and should we handle those cases differently ?
> 
> cannot execute binary file: Exec format error

I could only find references to "Exec format error".

Reply:
> > 32-bit system calling ldd against a 64-bit library
> 
> As long as we handle this case as well without regression, I am ok with this PR. I encourage others to test and review as well.

Hmm, nope, my guess was wrong:

```
# uname -a
Linux localhost.localdomain 6.1.10-1-default #1 SMP PREEMPT_DYNAMIC Mon Feb  6 07:27:26 UTC 2023 (2a0570b) i686 i686 i386 GNU/Linux
# file ./usr/lib64/libzscanner.so.4.0.0
./usr/lib64/libzscanner.so.4.0.0: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, BuildID[sha1]=45325b51127a2e895896db998a631b6b635d20df, stripped
# ldd !$
ldd ./usr/lib64/libzscanner.so.4.0.0
    not a dynamic executable
```

If "unable to execute binary" may occur in another context, I believe that in the worst case, `library_install` is unable to install the libraries referenced in the following lines.

Reply:
> > It would help to understand better what are the conditions for ldd to output "cannot execute binary file". Are they any conditions where the output matches "cannot execute binary file" but does NOT match "cannot execute binary file: Exec format error" and should we handle those cases differently ?
> > cannot execute binary file: Exec format error
> 
> I could only find references to "Exec format error".

Looking deeper, the bash code that prints this is here: https://git.savannah.gnu.org/cgit/bash.git/tree/execute_cmd.c#n6050

```c
internal_error (_("%s: cannot execute binary file: %s"), command, strerror (i));
```

IIUC the code above that line, it only prints that message if `i == ENOEXEC`, and glibc always prints "Exec format error" in that case: https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/gnu/errlist.h;h=e8416448da99051dc2bb8fe7c632346b90b0660c;hb=a704fd9a133bfb10510e18702f48a6a9c88dbbd5#l57